src/bin/common/solving_loop.ml

```diff
@@ -137,9 +137,16 @@ let main () =
         end;
       SAT.reset_refs ();
       let ftdn_env = FE.init_env used_context in
+      let hook_on_status status i =
+        Frontend.print_status status i;
+        match status with
+        | Timeout _ when not (Options.get_timelimit_per_goal ()) ->
+          exit_as_timeout ()
+        | _ -> ()
```

Comment:
I am in favor of either the current solution (unknown + error) or the `timeout` only, but I have a preference for the first one because it matches the standard

Reply:
My understanding is that the current behavior is actually not standard-compliant because we give two responses to a `(check-sat)` command where the standard only expects (at most; `success` can be skipped) one response per command.

(In any case, this is unrelated to this PR)

Reply:
I understand this as being a particular case of:

> Generally, once a solver completes its processing in response to a command, it should print to its regular output channel a 〈general_response〉

But it doesn't really matter; tools can process either output anyway (I had a glance at Why3's code for the SMT variant of Alt-Ergo and it looks like they currently don't parse timeouts at all there, so eh).